src/OrleansTestingHost/TestCluster.cs

```diff
@@ -455,7 +455,7 @@ public void InitializeClient()
                 }
                 catch (Exception)
                 {
-                    this.additionalSilos.AddRange(siloStartTasks.Where(t => t.Exception != null).Select(t => t.Result));
+                    this.additionalSilos.AddRange(siloStartTasks.Where(t => t.Exception == null).Select(t => t.Result));
```

Comment:
synced offline, we will add a logic to clean up silos which failed to start to enhance this fix.